### PR TITLE
Add a hook for ttkbootstrap that requires the tcl8 directory to be present.

### PR DESCRIPTION
**Pull request by Heribert17**
Tested for windows and with the newest version of cx_freeze that doesn't put the tkinter directories under tcltk.

### CODE REVIEW

File: cx_Freeze/hooks/__init__.py [hunk 1]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
     finder.include_package("tensorflow.python")
 
 
+def load_ttkbootstrap(finder: ModuleFinder, _: Module) -> None:
+    """The ttkbootstrap package needs the tcl8 directory.
+    This isn't copied by the Tkinter load hook, so we need an extra hook."""
+    folders = []
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:531]: Can you simplify this patch? If only one folder is to be included, you can remove this folder list.

File: cx_Freeze/hooks/__init__.py [hunk 2]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
...
+    tcltk = get_resource_file_path("bases", "tcltk", "")
+    if tcltk and tcltk.is_dir():
+        # manylinux wheels and macpython wheels store tcl/tk libraries
+        folders.append(("TCL_LIBRARY", list(tcltk.glob("tcl*"))[0]))
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:535]: source_path = list(tcltk.glob("tcl*"))[0])

File: cx_Freeze/hooks/__init__.py [hunk 3]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
...
+    else:
+        # Windows, MSYS2, Miniconda: collect the tcl/tk libraries
+        try:
+            tkinter = __import__("tkinter")
+        except (ImportError, AttributeError):
+            return
+        root = tkinter.Tk(useTk=False)
+        source_path = Path(root.tk.exprstring("$tcl_library"))
+        folders.append(("TCL_LIBRARY", source_path))
+    for _, source_path in folders:
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:545]: This for is unecessary...

File: cx_Freeze/hooks/__init__.py [hunk 4]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
...
+        # get the root tcl dir. If soure_path.name = tcl8.6 then tcl8
+        new_name = os.path.splitext(source_path.name)[0]
+        target_path = Path("lib", new_name)
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:548]: target_path = target_path = Path("lib", source_path.stem)

File: cx_Freeze/hooks/__init__.py [hunk 5]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
...
+        finder.include_files(
+            os.path.join(os.path.split(source_path)[0], new_name), target_path
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:550]: use Path operation here too...

File: cx_Freeze/hooks/__init__.py [hunk 6]
```diff
@@ -524,6 +525,32 @@ def load_tensorflow(finder: ModuleFinder, module: Module) -> None:
     finder.include_package("tensorflow.python")
 
 
+def load_ttkbootstrap(finder: ModuleFinder, _: Module) -> None:
```
Comment (marcelotduarte) [inline cx_Freeze/hooks/__init__.py:528]: Can you move this hook to a file ttkbootstrap.py?

**Review by marcelotduarte — DISMISSED** [review]

**Comment by marcelotduarte** [comment]
Analyzing better, I see that the tcl8 directory must be copied along with the others in load_tkinter (hooks/tkinter.py).
I'm going to do a complementary patch to make the tcl8 directory available on linux and macos.

**Comment by marcelotduarte** [comment]
You can test the patch in the latest development build:
`pip install --upgrade --pre --extra-index-url https://marcelotduarte.github.io/packages/ cx_Freeze`
For conda-forge the command is:
`conda uninstall cx_freeze`
`conda install -y --no-channel-priority -S -c https://marcelotduarte.github.io/packages/conda cx_Freeze`

**Comment by Heribert17** [comment]
Thank you very much. It works for my projects.

**Comment by marcelotduarte** [comment]
Thank you too.

**Comment by marcelotduarte** [comment]
[Release 6.14.6](https://github.com/marcelotduarte/cx_Freeze/releases/tag/6.14.6)

**Pull request closed without merging**